Update all hardhat plugin pages (#412)
matter-labs/zksync-web-era-docs: docs/api/hardhat/migrating-to-zksync.md

# Migration guide

> This guide shows you how to migrate an existing hardhat project to zkSync Era.

Reviewer: ```suggestion
This guide shows you how to migrate an existing Hardhat project to zkSync Era.
```

## Overview

> zkSync Era offers [multiple hardhat plugins](./plugins.md) with different features but in this guide, we are focused on only the required few needed to migrate your project to zkSync Era.

Reviewer: ```suggestion
zkSync Era offers [multiple Hardhat plugins](./plugins.md) with different features but in this guide, we are focused on only the required few needed to migrate your project to zkSync Era.
```

::: warning Non-default paths are not supported yet
- Contract files must be included in the `contracts` folder and deployment scripts must be included in the `deploy` folder.

> - Support for custom paths will be included in the future.

Reviewer: ```suggestion
- Support for custom paths will be included in the future.
```
Is this still true?

Author: So it's not live yet, but we'll soon release support to configure the path for the deployment scripts. 
So we can keep the message

:::

## Install dependencies

> Although zkSync Era is [compatible with Solidity and Vyper](../../dev/building-on-zksync/contracts/contract-development.md), the deployed bytecode and the deployment process is different from Ethereum or other EVM blockchains. So the first step is to install the compiler and deployer hardhat plugins:

Reviewer: ```suggestion
Although zkSync Era is [compatible with Solidity and Vyper](../../dev/building-on-zksync/contracts/contract-development.md), the deployed bytecode and the deployment process is different from Ethereum or other EVM blockchains. So the first step is to install the compiler and deployer Hardhat plugins:
```

import "@matterlabs/hardhat-zksync-solc";
```

> Networks on zkSync Era require two different URL endpoints: one for layer 1 (Ethereum or Goerli), and one for layer 2 (zkSync). This is how you would add the zkSync Era testnet to your list of networks in the `hardhat.config.ts`:

Reviewer: ```suggestion
Networks on zkSync Era require two different URL endpoints: one for layer 1 (Ethereum or Goerli), and one for layer 2 (zkSync). This is how you add the zkSync Era testnet to your list of networks in the `hardhat.config.ts`:
```

};
```
::: tip
Remember to add `zksync: false` to any other networks.
:::

> Finally, you will need to add the compiler options inside a `zksolc` or `zkvyper` property. Here is the minimal configuration for a Solidity project:

Reviewer: ```suggestion
Finally, add the compiler options inside a `zksolc` or `zkvyper` property. Here is the minimal configuration for a Solidity project:
```

```typescript
zksolc: {
  version: "1.3.8",
  compilerSource: "binary",
  settings: {},
},
```

> You can find more detailed config options in the [Solidity](./hardhat-zksync-solc.md) or [Vyper](./hardhat-zksync-vyper.md) plugins.

Reviewer: ```suggestion
For more advanced settings, check out the [Solidity](./hardhat-zksync-solc.md) or [Vyper](./hardhat-zksync-vyper.md) plugins.
```

:::

> To deploy your contracts you will need to use the `Deployer` class from the `hardhat-zksync-deploy`  plugin. This class takes care of all the specifics of [deploying contracts on zkSync](../../dev/building-on-zksync/contracts/contract-deployment.md). 

Reviewer: ```suggestion
To deploy your contracts you need to use the `Deployer` class from the `hardhat-zksync-deploy`  plugin. This class takes care of all the specifics of [deploying contracts on zkSync](../../dev/building-on-zksync/contracts/contract-deployment.md). 
```

```

> You will also need to use the `contract ABI` from the `artifacts-zk` folder to instantiate contracts.

Reviewer: ```suggestion
You also need to use the `contract ABI` from the `artifacts-zk` folder to instantiate contracts.
```
